src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -1179,9 +1180,29 @@ DropDefaultExpressionsAndMoveOwnedSequenceOwnerships(Oid sourceRelationId,
 	Oid ownedSequenceId = InvalidOid;
 	forboth_ptr_oid(columnName, columnNameList, ownedSequenceId, ownedSequenceIdList)
 	{
-		DropDefaultColumnDefinition(sourceRelationId, columnName);
+		HeapTuple columnTuple = SearchSysCacheAttName(sourceRelationId, columnName);
+		if (!HeapTupleIsValid(columnTuple))
+		{
+			ereport(ERROR, (errcode(ERRCODE_UNDEFINED_COLUMN),
+							errmsg("column \"%s\" of relation \"%s\" does not exist",
+								   columnName, get_rel_name(sourceRelationId))));
+		}
+
+		Form_pg_attribute columnForm = (Form_pg_attribute) GETSTRUCT(columnTuple);
+		AttrNumber columnAttrNumber = columnForm->attnum;
+
+		ReleaseSysCache(columnTuple);
 
-		/* column might not own a sequence */
+		if (ColumnDefaultsToNextVal(sourceRelationId, columnAttrNumber))
```

Comment:
Needs comment, why we only drop defaults for nextval sequences? Because we want to be able to use DEFAULT values in foreign keys when as foreign keys are evaluated on the shards.

I guess it should even be fine to keep nextvals as well, but could be confusing, so I'm fine dropping.